numba/core/typing/context.py

```diff
@@ -99,6 +119,66 @@ def match(self, py_func, args):
             if frame.args == args:
                 return frame
 
+    def lookup_resolve_cache(self, func, args, kws) -> "_ResolveCache":
+        """Lookup resolution cache for the given function type and argument
+        types.
+        """
+        if not self._stack or config.DISABLE_TYPEINFER_FAIL_CACHE:
+            # if callstack is empty, bypass fail_cache
+            return _ResolveCache()
+
+        def normalize_dict(obj):
+            if isinstance(obj, dict):
+                return tuple(kws.items())
+            return kws
+
+        def hashable(obj):
+            try:
+                hash(obj)
+            except TypeError:
+                return False
+            else:
+                return True
+
+        key = func, args, normalize_dict(kws)
+        if not hashable(key):
```

Comment:
A minor nit, but is there any reason not to use `isinstance(key, collections.abc.Hashable)` here?